lib/finch.ex

```diff
@@ -50,6 +50,28 @@ defmodule Finch do
     ]
   ]
 
+  @typedoc """
+  The `:name` provided to Finch in `start_link/1`.
+  """
+  @type name() :: atom()
+
+  @typedoc """
+  An HTTP request method represented as an atom or a String in all caps.
+
+  The following methods are supported: #{Enum.join(@methods, ", ")}.
```

Comment:
This makes me thing that any string methods are not supported, which is not true?

Reply:
Yes, in the current state, arbitrary string methods are not supported.

```
iex(2)> Finch.request(MyFinch, "WAT", "https://appmonet.com")
** (FunctionClauseError) no function clause matching in Finch.build_method/1

    The following arguments were given to Finch.build_method/1:

        # 1
        "WAT"

    Attempted function clauses (showing 2 out of 2):

        defp build_method(method) when method === "GET" or (method === "POST" or (method === "PUT" or (method === "PATCH" or (method === "DELETE" or (method === "HEAD" or method === "OPTIONS")))))
        defp build_method(method) when is_atom(method)

```

But I can't think of any good reason not to allow that flexibility, so I will open an issue to address that.